packages/drivers/odsp-driver/src/getUrlAndHeadersWithAuth.ts

```diff
@@ -3,40 +3,18 @@
  * Licensed under the MIT License.
  */
 
-import { assert } from "@fluidframework/core-utils/internal";
+import { assert } from "@fluidframework/core-utils";
 
-export function getUrlAndHeadersWithAuth(
-	url: string,
-	token: string,
-	forceAccessTokenViaAuthorizationHeader: boolean,
-): { url: string; headers: { [index: string]: string } } {
-	assert(token.length > 0, 0x936 /* should be token */);
-
-	if (!forceAccessTokenViaAuthorizationHeader) {
-		// Pass access token via query string: this will make request be treated as 'simple' request
-		// which does not require OPTIONS call as part of CORS check.
-		const urlWithAccessTokenInQueryString = new URL(url);
-		// IMPORTANT: Do not apply encodeURIComponent to token, param value is automatically encoded
-		// when set via URLSearchParams class
-		urlWithAccessTokenInQueryString.searchParams.set("access_token", token);
-		// ODSP APIs have a limitation that the query string cannot exceed 2048 characters.
-		// If the query string exceeds 2048, we have to fall back to sending the access token as a header, which
-		// has a negative performance implication as it adds a performance overhead.
-		// NOTE: URL.search.length value includes '?' symbol and it is unclear whether backend logic which enforces
-		// query length limit accounts for it. This logic errs on side of caution and includes that key in overall
-		// query length.
-		if (urlWithAccessTokenInQueryString.search.length <= 2048) {
-			return {
-				headers: {},
-				url: urlWithAccessTokenInQueryString.href,
-			};
-		}
+export function getHeadersWithAuth(
+	// eslint-disable-next-line @rushstack/no-new-null
+	authHeader: string | null,
+): { [index: string]: string } {
+	assert(authHeader !== null && authHeader.length > 0, 0x936 /* should be token */);
+	if (!authHeader || authHeader.length === 0) {
```

Comment:
You should choose one or the other - it does not make sense first to assert that authHeader is non-empty string, and then to have IF statement checking the same.
Also !authHeader checks for empty strings